### PR TITLE
Provide instructions on using and accessing example data 

### LINKED ISSUE #119: Provide example data for external users
We should provide some example data that external users can use if they need. This data should be publicly available already and preferably a smaller dataset. We could use some of the [publicly available data that is available from 10X](https://www.10xgenomics.com/resources/datasets?query=&page=1&configure%5Bfacets%5D%5B0%5D=chemistryVersionAndThroughput&configure%5Bfacets%5D%5B1%5D=pipeline.version&configure%5BhitsPerPage%5D=500). 

One approach to this would be to store this data in a public bucket on S3 and then fill in the `example_metadata.tsv` that we will provide with the paths to this data.

### PR DESCRIPTION
**Pull request by allyhawkins**
Closes #119, #145, and #146. 

Here I have included a README file for the `example` folder, providing instructions on how to access and download both the example input data and example output files as well as perform a test run using the input files. In the instructions I included the download links that can be used to access the files and then have instructed the users to create a metadata file for the example data, providing a table with an example of what to fill in for each column. 

In doing this I also updated the `example_metadata.tsv` to reflect only the columns that are required or optional and are listed in the section on preparing the metadata file in the external instructions. Specifically this meant removing the `library_method` column. I also added an example of a bulk sample, since we do mention that in the description of the column contents in the section. What I did not do was update the path to the example data on S3, since we want people to download the files locally and then test their setup rather directly using the files on S3. Correct me if I'm wrong on this and should correct this. 

However, in order to run the workflow using our config setup, I did input the path to the files on S3, and that metadata file that was used can be found on S3 in `s3://scpca-references/example-data/example_metadata.tsv`. That file contains just a single row with the path to the example outputs on S3. Let me know if you think this should be updated in anyway. 

Questions for reviewers: 
- Should we provide instructions on using either `wget` or `curl` to download the data? Or do we assume that users who want to use that will know how? 
- Is there anything that is unclear about the instructions? Or things that I missed?

Also to note that the zip file for the input files is still in the process of uploading to S3 but should be there soon and I used the `--acl public-read` to ensure the file was set to public.

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+## Example Files
```
Comment (jashapiro) [inline examples/README.md:1]: Making the title a bit more informative (and H1, why not)
```suggestion
# Example Metadata & Configuration Files for `scpca-nf`
```

File: examples/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+## Example Files 
+
+Here we provide an example of the [metadata file](../external-data-instructions.md#prepare-the-metadata-file) and [configuration file](../external-data-instructions.md#configuration-files) that you will need in order to setup a run of the `scpca-nf` workflow.
```
Comment (jashapiro) [inline examples/README.md:3]: Adding a bit of warning to set expectations. I don't love my wording, but I think we should include the sentiment.
```suggestion
This directory contains an example the [metadata file](../external-data-instructions.md#prepare-the-metadata-file) and [configuration file](../external-data-instructions.md#configuration-files) for the `scpca-nf` workflow. 
These files should be used as an example of formats and content, but note that the values in these files may not be applicable or sufficient to allow running `scpca-nf` to be used directly on your system.
```

File: examples/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+If you would like to test your configuration setup, after following the instructions outlined in the [external instructions](../external-data-instructions.md), you can do so by performing a test run with the example data that we have provided.
+
+We have an example 10X dataset from a [human glioblastoma donor that was processed using the 10X Genomics' Next GEM Single Cell 3' Reagent Kits v3.1](https://www.10xgenomics.com/resources/datasets/2-k-sorted-cells-from-human-glioblastoma-multiforme-3-v-3-1-3-1-standard-6-0-0) that is available for download. 
+The fastq files for this example data can be downloaded by navigating to https://s3.amazonaws.com/scpca-references/example-data/example_fastq_gbm_10x.zip. 
+
+Following download and unzipping of the fastq files, you will need to create a metadata file that looks like the following: 
+
+| scpca_run_id | scpca_library_id | scpca_sample_id | scpca_project_id | technology | seq_unit | files_directory | 
+| ------------ | ---------------- | --------------- | ---------------- | ---------- | -------- | --------------- | 
+| SCPCR999991 | SCPCL999991 | SCPCS999991 | SCPCP999991 | 10Xv3.1 | cell | path/to/example_fastq_files |
```
Comment (jashapiro) [inline examples/README.md:16]: I've been thinking about whether we really want to use SCPCX prefixes here. I think maybe not, as I think they may be confusing in this context. There is no requirement that they follow our ID format if they are just running on their own. 

I think we might want to change the example files as well, but that might warrant some more discussion.

I also made this look like a full path.

```suggestion
| run01 | library01 | sample01 | project01 | 10Xv3.1 | cell | /path/to/example_fastq_files | 
```

File: examples/README.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+We have an example 10X dataset from a [human glioblastoma donor that was processed using the 10X Genomics' Next GEM Single Cell 3' Reagent Kits v3.1](https://www.10xgenomics.com/resources/datasets/2-k-sorted-cells-from-human-glioblastoma-multiforme-3-v-3-1-3-1-standard-6-0-0) that is available for download. 
+The fastq files for this example data can be downloaded by navigating to https://s3.amazonaws.com/scpca-references/example-data/example_fastq_gbm_10x.zip. 
+
+Following download and unzipping of the fastq files, you will need to create a metadata file that looks like the following: 
+
+| scpca_run_id | scpca_library_id | scpca_sample_id | scpca_project_id | technology | seq_unit | files_directory | 
+| ------------ | ---------------- | --------------- | ---------------- | ---------- | -------- | --------------- | 
+| SCPCR999991 | SCPCL999991 | SCPCS999991 | SCPCP999991 | 10Xv3.1 | cell | path/to/example_fastq_files | 
+
+Be sure to enter the full path to where the fastq files are located in the `files_directory` column.
```
Comment (jashapiro) [inline examples/README.md:18]: Emphasis added.
```suggestion
Be sure to enter the **full** path to the directory containing the fastq files in the `files_directory` column.
```

File: examples/README.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  -profile <name of profile> \
+  --run_metafile <path/to/example_metadata.tsv>
+```
+
+**Note:** This setup assumes that you have already created a configuration file, `my_config.config` (or the filename of your choice) and have created and named a profile to use. 
+See the instructions on [creating a configuration file](../external-data-instructions.md#configuration-files) and [setting up a profile](../external-data-instructions.md#setting-up-a-profile-in-the-configuration-file) for more information.
+
+## Example output 
+
+You can download an example of the expected output files at https://s3.amazonaws.com/scpca-references/example-data/scpca_out.zip.
```
Comment (jashapiro) [inline examples/README.md:36]: I don't like bare links?
```suggestion
You can download an example of the expected output files here: [scpca_out.zip](https://s3.amazonaws.com/scpca-references/example-data/scpca_out.zip). 
```

File: examples/README.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+| SCPCR999991 | SCPCL999991 | SCPCS999991 | SCPCP999991 | 10Xv3.1 | cell | path/to/example_fastq_files | 
+
+Be sure to enter the full path to where the fastq files are located in the `files_directory` column.
+
+You can provide the path to the metadata file at the command line (here we name the file `example_metadata.tsv` but you can use the filename of your choice).
+The following command can then be used to test your configuration setup with the example data: 
+
+```
+nextflow run AlexsLemonade/scpca-nf \
+  -r v0.2.6 \
```
Comment (jashapiro) [inline examples/README.md:25]: Maybe we can leave the version out here, as it is not really important to this example, and  people should get the latest by default. 
```suggestion
```

**Review by jashapiro — COMMENTED** [review]
I think this looks good overall, thank you for adding it!

I had a number of minor comments throughout, some of which might require more discussion, in particular the question about what our example ids should look like.

File: examples/README.md (hunk 3 above)
Comment (allyhawkins) [reply examples/README.md:16]: I definitely like the changes you implemented here and would agree that we should probably update the files to have names more generalizable. That might be a change that we make in a separate PR though and we can leave it as is for now.

**Comment by allyhawkins** [comment]
I went ahead and updated the instructions based on @jashapiro's comments. In doing this I updated the link to the input fastq files to be the link directly provided by 10X, rather than the input files that we have stored. 

I also updated the ID's in the example table that I included. I would agree with Josh's original comment that we should update the `example_metadata.tsv` file to also utilize a different naming pattern than the `SCPCX` pattern but thought that change better fit in a separate PR/issue. 

One thing to note is that we should probably make sure the output data contains the same ID's that we are suggesting here, rather than the `SCPCX` id's as the folder names like they are now. If others agree, I will file an issue to track updating the output folder to reflect the naming chosen in this example table in the instructions here. 

Since Josh is out, going to re-route this PR to @jaclyn-taroni for review, but let me know if there is someone else I should have review.

File: examples/README.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+These files should be used as an example of formats and content, but note that the values in these files may not be applicable or sufficient to allow running `scpca-nf` to be used directly on your system.
+
+## Testing your setup with example data
+
+If you would like to test your configuration setup, after following the instructions outlined in the [external instructions](../external-data-instructions.md), you can do so by performing a test run with the example data that we have provided.
+
+We recommend using the example 10X dataset from a [human glioblastoma donor that was processed using the 10X Genomics' Next GEM Single Cell 3' Reagent Kits v3.1](https://www.10xgenomics.com/resources/datasets/2-k-sorted-cells-from-human-glioblastoma-multiforme-3-v-3-1-3-1-standard-6-0-0). 
+The fastq files for this example data can be downloaded from the following link: [Brain_Tumor_3p_fastqs.tar](https://cf.10xgenomics.com/samples/cell-exp/6.0.0/Brain_Tumor_3p/Brain_Tumor_3p_fastqs.tar).
```
Comment (jaclyn-taroni) [inline examples/README.md:11]: How big is this download? Should we warn people?

**Review by jaclyn-taroni — COMMENTED** [review]
It does appear that you addressed all of Josh's comments, but I had some of my own!

**Comment by allyhawkins** [comment]
Thanks for the review @jaclyn-taroni. I did a little bit of rearranging, making the beginning section about setting up the configuration file and making sure that you had the necessary pieces before proceeding on to using the example data. I also added a paragraph explaining each of the arguments for the `nextflow run` command and then a note about the size of the download for the input files. 

I added a line about needed the config file at the beginning and then changed the note that I had at the end to a warning. Do we need another warning at the beginning of the `testing your setup` section? I also thought about relocating the warning to the beginning of that section, but no matter where I put it, it still felt like it was slightly repetitive to the information before it. Let me know if you had something else in mind or if things are still unclear.

File: examples/README.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
+# Example Metadata & Configuration Files for `scpca-nf`
+
+## Set up configuration file
```
Comment (jaclyn-taroni) [inline examples/README.md:3]: What's currently here makes me think that these are going to be instructions for setting up a configuration file. That's not the case though. 

```suggestion
## Example files
```

File: examples/README.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+This directory contains an example [metadata file](../external-data-instructions.md#prepare-the-metadata-file) and [configuration file](../external-data-instructions.md#configuration-files) for the `scpca-nf` workflow. 
+These files should be used as an example of formats and content, but note that the values in these files may not be applicable or sufficient to allow running `scpca-nf` to be used directly on your system.
+Before following the instructions below, please ensure that you have already set up your own [configuration file](../external-data-instructions.md#configuration-files) and [created a profile](../external-data-instructions.md#setting-up-a-profile-in-the-configuration-file) to run `scpca-nf` on your system. 
+
+## Testing your setup with example data 
+
```
Comment (jaclyn-taroni) [inline examples/README.md:10]:
```suggestion

## Testing your setup with example data 

⚠️  Before following these instructions, please ensure that you have already set up your own [configuration file](../external-data-instructions.md#configuration-files) and [created a profile](../external-data-instructions.md#setting-up-a-profile-in-the-configuration-file) to run `scpca-nf` on your system. 

```

Comment (jaclyn-taroni) [reply examples/README.md:10]: Currently the warning is tucked as the last paragraph in a section I may or may not read if I've arrived at these docs with the purpose of testing my setup with example data.

File: examples/README.md [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+nextflow run AlexsLemonade/scpca-nf \
+  -config <path to config file>  \
+  -profile <name of profile> \
+  --run_metafile <path to metadata file>
+```
+
+Here, you will need to provide the relative path to the configuration file that you have setup after following the instructions on [creating a configuration file](../external-data-instructions.md#configuration-files). 
+You also will need to provide the name of the profile that you chose when creating a profile. 
+For the example configuration file that we provided, we used the profile name `cluster` and would indicate that we would like to use that profile at the command line with `-profile cluster`. 
+Finally, you will need to provide the path to the metadata file containing the associated metadata for the example data that you have set up following the instructions above.
```
Comment (jaclyn-taroni) [inline examples/README.md:37]:
```suggestion
```

File: examples/README.md [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+You also will need to provide the name of the profile that you chose when creating a profile. 
+For the example configuration file that we provided, we used the profile name `cluster` and would indicate that we would like to use that profile at the command line with `-profile cluster`. 
+Finally, you will need to provide the path to the metadata file containing the associated metadata for the example data that you have set up following the instructions above. 
+For more detailed information on setting up the metadata file for your own data, see instructions on [preparing the metadata file](../external-data-instructions.md#prepare-the-metadata-file). 
+
+:warning: These instructions are only intended to be used to test accurate set up of a configuration file. 
+This setup assumes that you have already created a configuration file and have created and named a profile to use. 
+See the instructions on [creating a configuration file](../external-data-instructions.md#configuration-files) and [setting up a profile](../external-data-instructions.md#setting-up-a-profile-in-the-configuration-file) for more information.
```
Comment (jaclyn-taroni) [inline examples/README.md:42]: I've already read the section once I get to this warning.

**Comment by allyhawkins** [comment]
@jaclyn-taroni I moved the warning to be at the beginning of the section. I interpreted your comment that it should be moved and not both at the beginning and the end of the section, let me know if I interpreted that incorrectly. I also reworded the description of the arguments, emphasizing when relative vs. full paths are necessary and adding in the link to the example file like you suggested.

**Review by jaclyn-taroni — APPROVED** [review]
Looks good! Thanks for the changes. I had one remaining comment that you can take or leave!

**Pull request merged**